### PR TITLE
Updates to Facebook Adapter

### PR DESCRIPTION
**Pull request by garypretty**
Contributes towards #3388 

- Update the Facebook Adapter to take adapter options by default, with an optional param for overriding the client used.
- Updated code docs and removed error / warning supression.
- Updated unit tests.

### CODE REVIEW

File: libraries/Adapters/Microsoft.Bot.Builder.Adapters.Facebook/FacebookAdapter.cs [hunk 1]
```diff
@@ -46,19 +50,21 @@ public class FacebookAdapter : BotAdapter, IBotFrameworkHttpAdapter
...
         /// Initializes a new instance of the <see cref="FacebookAdapter"/> class using an existing Facebook client.
         /// </summary>
-        /// <param name="facebookClient">A Facebook API interface.</param>
+        /// <param name="options">Options for the Facebook Adapter.</param>
         /// <param name="logger">The logger this adapter should use.</param>
-        /// <exception cref="ArgumentNullException"><paramref name="facebookClient"/> is null.</exception>
-        public FacebookAdapter(FacebookClientWrapper facebookClient, ILogger logger = null)
+        /// <param name="facebookClient">Client used to interact with the Facebook API.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="options"/> is null.</exception>
+        public FacebookAdapter(FacebookAdapterOptions options, ILogger logger = null, FacebookClientWrapper facebookClient = null)
```
Comment (mrivera-ms) [inline libraries/Adapters/Microsoft.Bot.Builder.Adapters.Facebook/FacebookAdapter.cs:64]: Does this break backward compatibility?

File: libraries/Adapters/Microsoft.Bot.Builder.Adapters.Facebook/FacebookEvents/FacebookResponseEvent.cs [hunk 2]
```diff
@@ -6,6 +6,9 @@
 
 namespace Microsoft.Bot.Builder.Adapters.Facebook.FacebookEvents
 {
+    /// <summary>
+    /// Represents in the incoming object received from Facebook and processed by the adapter.
```
Comment (mrivera-ms) [inline libraries/Adapters/Microsoft.Bot.Builder.Adapters.Facebook/FacebookEvents/FacebookResponseEvent.cs:10]: >    /// Represents in the [](start = 0, length = 25)

Should it be "Represents the incoming..."?

File: libraries/Adapters/Microsoft.Bot.Builder.Adapters.Slack/Microsoft.Bot.Builder.Adapters.Slack.csproj [hunk 3]
```diff
@@ -16,7 +16,7 @@
     <Description>Library for connecting bots with Slack.</Description>
     <Summary>This library implements C# classes for the Slack adapter.</Summary>
     <!-- The SlackAPI package isn't signed, so supress the warning. There seems to not be a way to supress this for ONLY SlackAPI. -->
-    <NoWarn>$(NoWarn),CS8002</NoWarn>
+    <NoWarn>$(NoWarn),CS8002,CS1591</NoWarn>
```
Comment (mrivera-ms) [inline libraries/Adapters/Microsoft.Bot.Builder.Adapters.Slack/Microsoft.Bot.Builder.Adapters.Slack.csproj:19]: >CS1591 [](start = 29, length = 6)

Why?

File: libraries/Adapters/Microsoft.Bot.Builder.Adapters.Facebook/FacebookAdapter.cs (hunk 1 above)
Comment (garypretty) [reply libraries/Adapters/Microsoft.Bot.Builder.Adapters.Facebook/FacebookAdapter.cs:64]: Yes, but we are OK as this is a preview package so this is part of the pre GA changes.

File: libraries/Adapters/Microsoft.Bot.Builder.Adapters.Facebook/FacebookEvents/FacebookResponseEvent.cs (hunk 2 above)
Comment (garypretty) [reply libraries/Adapters/Microsoft.Bot.Builder.Adapters.Facebook/FacebookEvents/FacebookResponseEvent.cs:10]: Thanks. Will update.

File: libraries/Adapters/Microsoft.Bot.Builder.Adapters.Slack/Microsoft.Bot.Builder.Adapters.Slack.csproj (hunk 3 above)
Comment (garypretty) [reply libraries/Adapters/Microsoft.Bot.Builder.Adapters.Slack/Microsoft.Bot.Builder.Adapters.Slack.csproj:19]: I'll check this again. I was getting build errors due to the Slack package not having code docs which I am addressing in a separate pr. Not sure why the suppression appears twice in project file. Will confirm tomorrow.

Comment (garypretty) [reply libraries/Adapters/Microsoft.Bot.Builder.Adapters.Slack/Microsoft.Bot.Builder.Adapters.Slack.csproj]: @mrivera-ms I have now updated this to remove the duplicate suppressions and leave just the individual property group that should be there https://github.com/microsoft/botbuilder-dotnet/blob/25d3c7d3aa9a968affd2fda4b747783aad1998d0/libraries/Adapters/Microsoft.Bot.Builder.Adapters.Slack/Microsoft.Bot.Builder.Adapters.Slack.csproj#L28

**Pull request merged**